sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/AnsiTypeCoercion.scala

```diff
@@ -120,6 +121,15 @@ object AnsiTypeCoercion extends TypeCoercionBase {
     case (_: TimestampType, _: DateType) | (_: DateType, _: TimestampType) =>
       Some(TimestampType)
 
+    case (t1: DayTimeIntervalType, t2: DayTimeIntervalType) =>
+      Some(DayTimeIntervalType(
+        IntervalUtils.minAnsiIntervalField(t1.startField, t2.startField),
```

Comment:
just `t1.startField.min(t2.startField)`?

Reply:
> just `t1.startField.min(t2.startField)`?

There is such a convenient API。。。Updated